### PR TITLE
bug 1553161: Restructure Balrog apps & dockerify tox.ini

### PR DESCRIPTION
**Pull request by mozbhearsum**
This gets Balrog much closer to following our Best Practices (https://github.com/mozilla-releng/releng-rfcs/blob/master/rfcs/0022-app-best-practices.md). The src/tests restructuring wasn't strictly necessary for that part, but its our new standard, so I thought it best to include it here.

It's probably easiest to view this as a set of changesets rather than a huge diff, as I've tried to group things together fairly logically. I _think_ things are relatively straightforward, despite the large amount of change. There's a few things I want to highlight to make sure they're not overlooked though:
* I'm removing `api-tests` which we've never run (they don't contain any value not covered by other tests).
* I've moved as much config as possible into `pyproject.toml` and `tox.ini`
* I'm volume mounting anything that gets created inside a Docker image to avoid root-owned files on the host system (.tox, .hypothesis, etc.)
* Coverage will appear to be going down, but it's actually staying the same for things that matter. What has changed is that we're no longer considering the tests as something that should be checked by code coverage.
* The images we built after merging to master _should_ work (`docker-compose up` works fine locally), but we should be careful to test them well, as the production Dockerfiles aren't used as part of local development/testing.

I'd like to follow-up at some point later to replace all of the `scripts` and `uwsgi` parts with proper `entry_points` that exist in the `auslib` module -- but I think that's overreaching a bit at the moment.

### CODE REVIEW

**Comment by mozbhearsum** [comment]
(The failure in the most recent push is a 502 during frontend tests - nothing that's impacted by the changes here.)

**Comment by mozbhearsum** [comment]
> Wow! Sure thing! It may take a day or two - want to properly go through the changes!

Take your time. Let me know if anything is unclear.

File: uwsgi/admin.wsgi [hunk 1]
```diff
@@ -90,14 +89,14 @@ try:
...
-    # fmt: on
     }
+    # fmt: on
     for bucket in buckets.values():
         blob = bucket.blob("startuptest")
         blob.upload_from_string("startuptest", content_type="text/plain")
 except (ValueError, Forbidden) as e:
     if not LOCALDEV:
-        if isinstance(e, Forbidden) or (isinstance(e, ValueError) and"Anonymous credentials" not in e.args[0]):
+        if isinstance(e, Forbidden) or (isinstance(e, ValueError) and "Anonymous credentials" not in e.args[0]):
```
Comment (rail) [inline uwsgi/admin.wsgi]: Awwww.

File: .taskcluster.yml [hunk 2]
```diff
@@ -121,15 +133,35 @@ tasks:
             routes: []
             payload:
               maxRunTime: 1200
-              image: "mozillareleases/python-test-runner"
+              image: "python:2.7"
```
[the remark this answers is not in the document]
Comment (mozbhearsum) [reply .taskcluster.yml]: This is for the `balrogclient`, which is still run through Python 2.7 We're also testing on 3.7 (line 158), so I _think_ we're okay for now.

I agree we should test this, and the other parts of the app, on 3.8 in the near future though.

File: client/pyproject.toml [hunk 3]
```diff
@@ -19,3 +19,10 @@ exclude = '''
   | ui
 )/
 '''
+
+[tool.isort]
+multi_line_output = 3
+include_trailing_comma = true
+force_grid_wrap = 0
+use_parentheses = true
+line_length = 160
```
[the remark this answers is not in the document]
Comment (mozbhearsum) [reply client/pyproject.toml]: This is just carrying over what we do for the main Balrog app...I actually want to pull it back to 100 at some point. But not in this PR :)

File: uwsgi/admin.wsgi (hunk 1 above)
Comment (mozbhearsum) [reply uwsgi/admin.wsgi]: Believe it or not, this was not an error. AFAICT this functions the same before and after.

File: .taskcluster.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mozbhearsum) [reply .taskcluster.yml]: TIL that we already migrated Balrog submission to 3.7. I'm going to drop this 2.7 block.

**Pull request merged**